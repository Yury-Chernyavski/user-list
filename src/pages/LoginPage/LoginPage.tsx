import React, { FC } from "react";
import { LoginForm } from "../../components";
import { Container } from "../../theme/components";

export const LoginPage: FC = () => {
  return (
    <Container>
      <LoginForm />
    </Container>
  );
};
