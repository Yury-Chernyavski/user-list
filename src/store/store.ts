import { combineReducers, configureStore } from "@reduxjs/toolkit";
import usersSlice from "./reducers/usersSlice";

const rootReducer = combineReducers({
  users: usersSlice,
});

const store = configureStore({
  reducer: rootReducer,
})

export type RootSate = ReturnType<typeof rootReducer>;
export type AppDispatch = typeof store.dispatch;
export default store;

