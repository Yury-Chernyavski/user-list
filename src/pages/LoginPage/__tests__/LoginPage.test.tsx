import { render } from "@testing-library/react";
import { LoginPage } from "../LoginPage";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../../store/store";


describe("MainHeader", () => {
  it("Should render the LoginPage component", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <LoginPage />
        </BrowserRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
