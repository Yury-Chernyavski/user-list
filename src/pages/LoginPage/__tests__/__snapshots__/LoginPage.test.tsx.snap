// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader Should render the LoginPage component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gswONk kJchHL"
      >
        <div
          class="sc-dkrGBB cvJVRC"
        >
          <h1
            class="sc-hLBatu QoYOU"
          >
            Log in
          </h1>
          <form
            data-testid="Login-form"
          >
            <input
              class="sc-bcXGCL ibUrKg"
              placeholder="Email"
              type="email"
              value=""
            />
            <input
              class="sc-bcXGCL ibUrKg"
              placeholder="Password"
              type="password"
              value=""
            />
            <button
              class="sc-eDvShL hKbctB primary"
              type="submit"
            >
              Login
            </button>
            <p
              class="sc-jSVask jnHlof"
            >
              You 
              don't
               have an account? 
              <a
                href="/register"
              >
                Sing up
              </a>
            </p>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gswONk kJchHL"
    >
      <div
        class="sc-dkrGBB cvJVRC"
      >
        <h1
          class="sc-hLBatu QoYOU"
        >
          Log in
        </h1>
        <form
          data-testid="Login-form"
        >
          <input
            class="sc-bcXGCL ibUrKg"
            placeholder="Email"
            type="email"
            value=""
          />
          <input
            class="sc-bcXGCL ibUrKg"
            placeholder="Password"
            type="password"
            value=""
          />
          <button
            class="sc-eDvShL hKbctB primary"
            type="submit"
          >
            Login
          </button>
          <p
            class="sc-jSVask jnHlof"
          >
            You 
            don't
             have an account? 
            <a
              href="/register"
            >
              Sing up
            </a>
          </p>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
